test/tests/serial/tests.bats

```diff
@@ -182,23 +182,15 @@ function teardown_file() { _default_teardown ; }
   assert_success
 }
 
-@test "system: /var/log/mail/mail.log is error free" {
-  _run_in_container grep 'non-null host address bits in' /var/log/mail/mail.log
-  assert_failure
-  _run_in_container grep 'mail system configuration error' /var/log/mail/mail.log
-  assert_failure
-  _run_in_container grep ': error:' /var/log/mail/mail.log
-  assert_failure
-  _run_in_container grep -i 'is not writable' /var/log/mail/mail.log
-  assert_failure
-  _run_in_container grep -i 'permission denied' /var/log/mail/mail.log
-  assert_failure
-  _run_in_container grep -i '(!)connect' /var/log/mail/mail.log
-  assert_failure
-  _run_in_container grep -i 'using backwards-compatible default setting' /var/log/mail/mail.log
-  assert_failure
-  _run_in_container grep -i 'connect to 127.0.0.1:10023: Connection refused' /var/log/mail/mail.log
-  assert_failure
+@test "system: Mail log is error free" {
+  _service_log_should_not_contain_string 'mail' 'non-null host address bits in'
+  _service_log_should_not_contain_string 'mail' 'mail system configuration error'
+  _service_log_should_not_contain_string 'mail' ': error:'
+  _service_log_should_not_contain_string 'mail' 'is not writable'
+  _service_log_should_not_contain_string 'mail' 'permission denied'
+  _service_log_should_not_contain_string 'mail' '\(!\)connect'
+  _service_log_should_not_contain_string 'mail' 'using backwards-compatible default setting'
+  _service_log_should_not_contain_string 'mail' 'connect to 127.0.0.1:10023: Connection refused'
```

Comment:
Could probably retrieve the log and use `refute_output --partial` instead?

Then again some of the statements were relying on `grep -i`, so they potentially match multiple or casing may not be accurate 🤷‍♂️ Bit harder with failure assertions to determine 😓 (_although in this case I think I added context in the vmail-id test PR_)

**EDIT:** Feedback given on that related snippet. A future PR would replace this in favour of the `vmail-id` one. Tracking for that is in https://github.com/docker-mailserver/docker-mailserver/issues/3554

Reply:
It's probably a good idea to solve this in a PR that focuses on #3554